sources/Interop/Advapi32/um/wincrypt/Advapi32.Manual.cs

```diff
@@ -0,0 +1,5951 @@
+
+        // ((LPCSTR) 11)
+        public const int CERT_STORE_PROV_COLLECTION = 11;
+
+        // ((LPCSTR) 12)
+        public const int CERT_STORE_PROV_SYSTEM_REGISTRY_A = 12;
+
+        // ((LPCSTR) 13)
+        public const int CERT_STORE_PROV_SYSTEM_REGISTRY_W = 13;
+
+        public const int CERT_STORE_PROV_SYSTEM_REGISTRY = CERT_STORE_PROV_SYSTEM_REGISTRY_W;
+
+        // ((LPCSTR) 14)
+        public const int CERT_STORE_PROV_PHYSICAL_W = 14;
+
+        public const int CERT_STORE_PROV_PHYSICAL = CERT_STORE_PROV_PHYSICAL_W;
+
+        // ((LPCSTR) 15)
+        public const int CERT_STORE_PROV_SMART_CARD_W = 15;
+
+        public const int CERT_STORE_PROV_SMART_CARD = CERT_STORE_PROV_SMART_CARD_W;
+
+        // ((LPCSTR) 16)
+        public const int CERT_STORE_PROV_LDAP_W = 16;
+
+        public const int CERT_STORE_PROV_LDAP = CERT_STORE_PROV_LDAP_W;
+
+        // ((LPCSTR) 17)
+        public const int CERT_STORE_PROV_PKCS12 = 17;
+
+        public static ReadOnlySpan<sbyte> sz_CERT_STORE_PROV_FILENAME => sz_CERT_STORE_PROV_FILENAME_W;
+
+        public static ReadOnlySpan<sbyte> sz_CERT_STORE_PROV_SYSTEM => sz_CERT_STORE_PROV_SYSTEM_W;
+
+        public static ReadOnlySpan<sbyte> sz_CERT_STORE_PROV_SYSTEM_REGISTRY => sz_CERT_STORE_PROV_SYSTEM_REGISTRY_W;
+
+        public static ReadOnlySpan<sbyte> sz_CERT_STORE_PROV_PHYSICAL => sz_CERT_STORE_PROV_PHYSICAL_W;
+
+        public static ReadOnlySpan<sbyte> sz_CERT_STORE_PROV_SMART_CARD => sz_CERT_STORE_PROV_SMART_CARD_W;
+
+        public static ReadOnlySpan<sbyte> sz_CERT_STORE_PROV_LDAP => sz_CERT_STORE_PROV_LDAP_W;
+
+        public const int CERT_SYSTEM_STORE_CURRENT_USER = CERT_SYSTEM_STORE_CURRENT_USER_ID << CERT_SYSTEM_STORE_LOCATION_SHIFT;
+
+        public const int CERT_SYSTEM_STORE_LOCAL_MACHINE = CERT_SYSTEM_STORE_LOCAL_MACHINE_ID << CERT_SYSTEM_STORE_LOCATION_SHIFT;
+
+        public const int CERT_SYSTEM_STORE_CURRENT_SERVICE = CERT_SYSTEM_STORE_CURRENT_SERVICE_ID << CERT_SYSTEM_STORE_LOCATION_SHIFT;
+
+        public const int CERT_SYSTEM_STORE_SERVICES = CERT_SYSTEM_STORE_SERVICES_ID << CERT_SYSTEM_STORE_LOCATION_SHIFT;
+
+        public const int CERT_SYSTEM_STORE_USERS = CERT_SYSTEM_STORE_USERS_ID << CERT_SYSTEM_STORE_LOCATION_SHIFT;
+
+        public const int CERT_SYSTEM_STORE_CURRENT_USER_GROUP_POLICY = CERT_SYSTEM_STORE_CURRENT_USER_GROUP_POLICY_ID << CERT_SYSTEM_STORE_LOCATION_SHIFT;
+
+        public const int CERT_SYSTEM_STORE_LOCAL_MACHINE_GROUP_POLICY = CERT_SYSTEM_STORE_LOCAL_MACHINE_GROUP_POLICY_ID << CERT_SYSTEM_STORE_LOCATION_SHIFT;
+
+        public const int CERT_SYSTEM_STORE_LOCAL_MACHINE_ENTERPRISE = CERT_SYSTEM_STORE_LOCAL_MACHINE_ENTERPRISE_ID << CERT_SYSTEM_STORE_LOCATION_SHIFT;
+
+        public const int CERT_SYSTEM_STORE_LOCAL_MACHINE_WCOS = CERT_SYSTEM_STORE_LOCAL_MACHINE_WCOS_ID << CERT_SYSTEM_STORE_LOCATION_SHIFT;
+
+        public const string CERT_GROUP_POLICY_SYSTEM_STORE_REGPATH =
```

Comment:
These are (mostly) just macro functions that weren't ported, right?

Reply:
I'm a bit confused about how your question relates to the code you highlighted. There are many `#define`s in `wincrypt.h` that are basically simple math expressions (like the one you highlighted). However, I did not port any macro functions at all throughout this PR (i.e., anything with `#define F(X)`.

Reply:
The code you highlighted is defined thusly:
```c
#define CERT_SYSTEM_STORE_LOCAL_MACHINE_WCOS  \
    (CERT_SYSTEM_STORE_LOCAL_MACHINE_WCOS_ID << \
        CERT_SYSTEM_STORE_LOCATION_SHIFT)
```

Reply:
For whatever reason, the "Conversation" tab puts this on the wrong line.

If you goto the "Files" tab, it is on L5919: https://github.com/terrafx/terrafx.interop.windows/pull/37/files/435c24606296062bd557b116d453c517e1e042cb#diff-d03360204625b1a7168b85116fe6c24bR5919

Reply:
Ah, yes, those are all the remaining `#define`s. The PowerShell script creates comments for every `#define` that couldn't be mapped automatically. I then go through the list and convert any that I feel like I can manage. Most of the leftovers are "functions."